safe_prime_generator.go

```diff
@@ -0,0 +1,293 @@
+// Copyright 2011 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// The code is the original Go implementation of rand.Prime optimized for
+// generating safe (Sophie Germain) primes.
+// A safe prime is a prime number of the form 2p + 1, where p is also a prime.
+
```

Comment:
Yeah, otherwise it complains 
> package comment should be of the form "Package paillier ..."

I don't want to comment what the `paillier` package is for. I want to give a brief information what was changed comparing to Go implementation.